File: 0383-ransom-note/0383-ransom-note.js
/**
 * @param {string} ransomNote
 * @param {string} magazine
 * @return {boolean}
 */
var canConstruct = function(ransomNote, magazine) {
    let result = true;
    let mz = magazine;

    for (let char of ransomNote) {
        if(mz.indexOf(char) === -1) {
            result = false;
            break;
        }
        mz = mz.replace(char, '');
    }
    return result;
};

File: 0383-ransom-note/0383-ransom-note.java
class Solution {
    public boolean canConstruct(String ransomNote, String magazine) {
        Map<Character, Integer> map = new HashMap<>();
        int[] mCount = new int[26];
        int[] rCount = new int[26];

        for (int i = 0; i < magazine.length(); i++) {
            char mc = magazine.charAt(i);
            mCount[mc - 'a']++;
        }

        for (int i = 0; i < ransomNote.length(); i++) {
            char rc = ransomNote.charAt(i);
            rCount[rc - 'a']++;
        }

        for (int i = 0; i < 26; i++) {
            if (mCount[i] < rCount[i])
                return false;
        }

        return true;
    }
}